pkg/validator/interface.go

```diff
@@ -0,0 +1,18 @@
+package validator
+
+import (
+	"github.com/uvio-network/apiserver/pkg/validator/postvalidator"
+	"github.com/uvio-network/apiserver/pkg/validator/votevalidator"
+)
+
+// Interface provides cross resource validation for storage specific
+// implementations. For instance the vote storage may need to use the post
+// storage in order to verify the integrity of vote objects. At the same time
+// the post storage may need to use the vote storage in order to verify the
+// integrity of the post objects. The abstraction of the validator interface
+// allows the validation of business logic that is cross dependent on multiple
+// resources in either direction.
+type Interface interface {
+	Post() postvalidator.Interface
+	Vote() votevalidator.Interface
+}
```

Comment:
My earlier idea for the post and vote management turned out to be insufficient. I would now like to manage posts and vote more independently according to their respective lifecycles and let the clients hydrate and annotate information where they may be required.

This makes a lot of sense because the data associated with claims and the value associated with votes are each decoupled by intended user experience and underlying infrastructure. One claim is usually managed once, while many votes are managed separately across the lifetime of the associated claim. And then, the data storage for claims and the value storage for votes are backed by different infrastructure.

So the clients will be responsible for fetching claim data and vote value in order to present them in the intended ways.